ui/ui-components/utils/appContext.ts

```diff
@@ -1,11 +1,34 @@
 import { GetServerSidePropsContext, NextPageContext } from "next";
 import { AppContext } from "next/app";
 
+export type NextContextType =
+  | "AppContext"
+  | "NextPageContext"
+  | "ServerSidePropsContext";
+
 function isAppContext(ctx: unknown): ctx is AppContext {
   return !!ctx.hasOwnProperty("ctx");
 }
+
+function getNextContextType(ctx: unknown): NextContextType {
```

Comment:
Nitpick: `unknown` here could be `AppContext | NextPageContext | GetServerSidePropsContext`